build/build.c

```diff
@@ -172,9 +177,70 @@ rpmRC doScript(rpmSpec spec, rpmBuildFlags what, const char *name,
     return rc;
 }
 
-static rpmRC buildSpec(BTA_t buildArgs, rpmSpec spec, int what)
+static int doBuildRequires(rpmSpec spec, int test)
+{
+    StringBuf sb_stdout = NULL;
+    int outc;
+    ARGV_t output = NULL;
+
+    int rc = 1; /* assume failure */
+
+    if (!spec->buildrequires) {
+	rc = RPMRC_OK;
+	goto exit;
+    }
+
+    if ((rc = doScript(spec, RPMBUILD_BUILDREQUIRES, "%generate_buildrequires",
+		       getStringBuf(spec->buildrequires), test, &sb_stdout)))
+	goto exit;
+
+    /* add results to requires of the srpm */
+    argvSplit(&output, getStringBuf(sb_stdout), "\n\r");
+    outc = argvCount(output);
+
+    if (!outc) {
+	goto exit;
+    }
+
+    for (int i = 0; i < outc; i++) {
+	parseRCPOT(spec, spec->sourcePackage, output[i], RPMTAG_REQUIRENAME, 0, 0, addReqProvPkg, NULL);
+    }
+
+    rpmdsPutToHeader(
+	*packageDependencies(spec->sourcePackage, RPMTAG_REQUIRENAME),
+	spec->sourcePackage->header);
+
+    parseRCPOT(spec, spec->sourcePackage,
+	       "rpmlib(DynamicBuildRequires) = 4.15.0-1",
+	       RPMTAG_PROVIDENAME, 0, RPMSENSE_FIND_PROVIDES | RPMSENSE_RPMLIB,
+	       addReqProvPkg, NULL);
+    rc = RPMRC_MISSINGBUILDREQUIRES;
+
+ exit:
+    freeStringBuf(sb_stdout);
+    free(output);
+    return rc;
+}
+
+static rpmRC doCheckBuildRequires(rpmts ts, rpmSpec spec, int test)
+{
+    rpmRC rc = RPMRC_OK;
+    rpmps ps = rpmSpecCheckDeps(ts, spec);
+
+    if (ps) {
+	rpmlog(RPMLOG_ERR, _("Failed build dependencies:\n"));
+	rpmpsPrint(NULL, ps);
+    }
+    if (ps != NULL)
+	rc = RPMRC_MISSINGBUILDREQUIRES;
```

Comment:
This latter (ps != NULL) is redundant, just move the rc setting to the if (ps) case above. Sorry for missing on the previous rounds.